spyder/plugins/editor/utils/autosave.py

```diff
@@ -174,7 +174,8 @@ def try_recover_from_autosave(self):
         the pid files.
         """
         files_to_recover, pidfiles = self.get_files_to_recover()
-        dialog = RecoveryDialog(files_to_recover, parent=self.editor)
+        parent = self.editor if running_under_pytest() else self.editor.main
+        dialog = RecoveryDialog(files_to_recover, parent=parent)
```

Comment:
Out of curiosity, why this change?

Reply:
The initial change was to use the `MainWindow` instead of the `Editor` to properly set the `styleSheet` for the Widget (before this the dialog always was shown using the light theme). However, when running the test suit we use a mock for the `MainWindow` that inherits from a 
`QObject`, and since the dialog needs a QWidget as parent, the tests suit was failing here. Any suggestions to handle this are greatly appreciated :)

Reply:
I see, I had this problem that the parent should be a `QWidget` before; unfortunately, I don't know of a way around this, so I'm afraid I can't help you.

Reply:
No problem, thanks for looking into this :+1: